### PR TITLE
Integrate Mantaray into XLML

### PR DESCRIPTION
**Pull request by wenxindongwork**
# Description

Enable running Mantaray jobs on XLML Airflow. 

Mantaray is a benchmarking library. The Mantaray DAG first downloads the Mantaray wheel and the Mantaray benchmarks from GCS, and then launches the benchmarks. 

# Tests

Tested running Mantaray jobs e2e on the `cloud-ml-auto-solutions-dev` Composer Environment.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/mantaray/mantaray.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+import datetime
+from airflow import models
+from xlml.utils import mantaray
+import yaml
+
+# Download xlml_jobs.yaml from GCS
+xlml_jobs_yaml = mantaray.load_file_from_gcs(
+    "gs://us-central1-ml-automation-s-bc954647-bucket/mantaray/xlml_jobs/xlml_jobs.yaml"
```
Comment (allenwang28) [inline dags/mantaray/mantaray.py:25]: should this bucket path be hard coded or configurable?

File: xlml/utils/mantaray.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        f"pip install mantaray-0.1-py2.py3-none-any.whl",
+        f"python xlml_jobs/{workload_file_name}",  # Run the workload
+    )
+    hook = SubprocessHook()
+    result = hook.run_command(
+        ["bash", "-c", ";".join(cmds)],
+    )
+    assert (
+        result.exit_code == 0
+    ), f"XPK command failed with code {result.exit_code}"
```
Comment (allenwang28) [inline xlml/utils/mantaray.py:60]: should this say `f"MantaRay failed with code {result.exit_code}:`?

**Review by allenwang28 — COMMENTED** [review]
Nice work! Some minor comments

**Review by allenwang28 — COMMENTED** [review]
Nice work! Some minor comments

File: xlml/utils/mantaray.py (hunk 2 above)
Comment (wenxindongwork) [reply xlml/utils/mantaray.py]: fixed typo, thank you for catching it!

File: dags/mantaray/mantaray.py (hunk 1 above)
Comment (wenxindongwork) [reply dags/mantaray/mantaray.py:25]: I think we must use this path,  this path belongs to the storage bucket of the`ml-automation-solutions` composer environment. New jobs must be uploaded to this bucket in order to trigger DAG (re)creation.

Comment (allenwang28) [reply dags/mantaray/mantaray.py:25]: Ok that makes sense! Can you add a comment that clarifies the importance of this bucket?

**Review by allenwang28 — APPROVED** [review]

File: dags/mantaray/mantaray.py (hunk 1 above)
Comment (wenxindongwork) [reply dags/mantaray/mantaray.py:27]: Thank you for the suggestion, added!

File: dags/mantaray/mantaray.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+import datetime
+from airflow import models
+from xlml.utils import mantaray
+import yaml
+from dags import composer_env
+
+# Skip running this script in unit test because gcs loading will fail.
+if composer_env.is_prod_env() or composer_env.is_dev_env():
```
Comment (allenwang28) [inline dags/mantaray/mantaray.py:25]: nit, is it possible to print out in the else branch that we aren't running this since this is not in the right zone?

Comment (wenxindongwork) [reply dags/mantaray/mantaray.py:25]: sounds good!

**Pull request merged**